### PR TITLE
fix: simplify plugins comparison - master

### PR DESCRIPTION
**Pull request by phiz71**
**Issue**

https://gravitee.atlassian.net/browse/APIM-4080

**Description**

Merge the SecretProviderPluginComparator logic into the PluginComparator to respect `Comparator` contract.

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `4.0.3-apim-4080-fix-plugin-comparator-master-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/plugin/gravitee-plugin/4.0.3-apim-4080-fix-plugin-comparator-master-SNAPSHOT/gravitee-plugin-4.0.3-apim-4080-fix-plugin-comparator-master-SNAPSHOT.zip)

### CODE REVIEW

File: gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/internal/AbstractPluginEventListener.java [hunk 1]
```diff
@@ -127,17 +126,39 @@ private void deployPlugin(Plugin plugin, Map<String, List<Plugin>> resolvedDepen
...
-            Integer pos2 = pluginPriority.indexOf(o2.type());
+            int pos1 = pluginPriority.indexOf(o1.type());
+            int pos2 = pluginPriority.indexOf(o2.type());
 
             if (pos1 >= 0) {
                 if (pos2 >= 0) {
+                    if (pos1 == pos2) {
+                        if (
+                            loadFirst != null &&
+                            o1.manifest().type().equalsIgnoreCase(SECRET_PROVIDER) &&
```
Comment (guillaumelamirand) [inline gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/internal/AbstractPluginEventListener.java:147]: AbstractPluginEventListener should be agnostic of this kind of details. 
Could you explain the solution according to the initial exception ?

Comment (benoitgravitee) [inline gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/internal/AbstractPluginEventListener.java:147]: You probably want to test the plugin type first, then make a condition on `loadFirst`. If it is not a secret provider plugin then `loadFirst` won't matter, value is bound to this specific type of plugin.

```pseudo code
if pluginType == "secret provider" and loadFirst is not null or empty 
  if o1.id == loadFirst
   return -1
  else if o2.id == loadFirst
  return 1
end if
return comparison of positions
```

This way you respect the contract and make it clear why you compare `loadFirst`

But let's wait for the above comment to be answered

Comment (benoitgravitee) [reply gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/internal/AbstractPluginEventListener.java:147]: After discussion we keep things as is

Comment (benoitgravitee) [reply gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/internal/AbstractPluginEventListener.java:147]: This has then to be changed

**Review by benoitgravitee — APPROVED** [review]

**Pull request merged**